File: python/7 kyu/435.py
# https://www.codewars.com/kata/553f01db29490a69ff000049/train/python

# Passed

def validate_sequence(sequence):
    diff_set = {sequence[idx + 1] - sequence[idx] for idx in range(0, len(sequence) - 1)}
    result = len(diff_set) == 1
    return result

output = validate_sequence([1, 2, 3, 4, 5, 6, 7, 8, 9])
print(output)

File: nodejs/7 kyu/435.js
// https://www.codewars.com/kata/64b8c6c09416795eb9fbdcbf/train/javascript

// Passed

function repSet(n) {
  if (n === 0) {
    return [];
  }

  let result = [];
  for (let currentNum = 0; currentNum < n; currentNum++) {
    let currentNumSet = repSet(currentNum);
    result.push(currentNumSet);
  }

  return result;
}

const output = repSet(3);
console.log(output);